Framework/DataHandling/src/LoadSwans.cpp

```diff
@@ -0,0 +1,298 @@
+    loadInst->setProperty("RewriteSpectraMap",
+                          Mantid::Kernel::OptionalBool(true));
+    loadInst->execute();
+  } catch (...) {
+    g_log.information("Cannot load the instrument definition.");
+  }
+}
+
+/**
+ * Place the detector in space according to the distance and angle
+ * Needs in the IDF Parameters file the entries:
+ * detector-name, detector-sample-distance
+ * Also the metadata file has to have the rotation angle value
+ */
+void LoadSwans::placeDetectorInSpace() {
+
+  std::string componentName =
+      m_ws->getInstrument()->getStringParameter("detector-name")[0];
+  const double distance = static_cast<double>(
+      m_ws->getInstrument()->getNumberParameter("detector-sample-distance")[0]);
+  // Make the angle negative to accomodate sense of rotation.
+  const double angle = -m_ws->run().getPropertyValueAsType<double>("angle");
+
+  g_log.information() << "Moving detector " << componentName << " " << distance
+                      << " meters and " << angle << " degrees." << std::endl;
+
+  LoadHelper helper;
+  constexpr double deg2rad = M_PI / 180.0;
+  V3D pos = helper.getComponentPosition(m_ws, componentName);
+  double angle_rad = angle * deg2rad;
+  V3D newpos(distance * sin(angle_rad), pos.Y(), distance * cos(angle_rad));
+  helper.moveComponent(m_ws, componentName, newpos);
+
+  // Apply a local rotation to stay perpendicular to the beam
+  const V3D axis(0.0, 1.0, 0.0);
+  Quat rotation(angle, axis);
+  helper.rotateComponent(m_ws, componentName, rotation);
+}
+
+/**
+ * Load the data into a map. The map is indexed by pixel id (0 to 128*128-1 =
+ * m_detector_size)
+ * The map values are the events TOF
+ * @returns the map of events indexed by pixel index
+ */
+std::map<uint32_t, std::vector<uint32_t>> LoadSwans::loadData() {
+
+  std::string filename = getPropertyValue("FilenameData");
+  std::ifstream input(filename, std::ifstream::binary | std::ios::ate);
+  input.seekg(0);
+
+  m_ws->initialize(m_detector_size, 1, 1);
+
+  std::map<uint32_t, std::vector<uint32_t>> eventMap;
+
+  while (input.is_open()) {
+    if (input.eof())
+      break;
+    uint32_t tof = 0;
+    // input.read((char *)&tof, sizeof(tof));
+    input.read(reinterpret_cast<char *>(&tof), sizeof(tof));
+    tof -= static_cast<uint32_t>(1e9);
+    tof = static_cast<uint32_t>(tof * 0.1);
+
+    uint32_t pos = 0;
+    // input.read((char *)&pos, sizeof(pos));
+    input.read(reinterpret_cast<char *>(&pos), sizeof(pos));
+    if (pos < 400000) {
+      g_log.warning() << "Detector index invalid: " << pos << std::endl;
+      continue;
+    }
+    pos -= 400000;
+    eventMap[pos].push_back(tof);
+  }
+
+  return eventMap;
+}
+
+/**
+ * Load metadata file witch to date is just a line of of double values
+ * Parses this file and put it into a vector
+ * @return vector with the file contents
+ */
+std::vector<double> LoadSwans::loadMetaData() {
+
+  std::vector<double> metadata;
```

Comment:
Change the type to `std::array<double,6>` and check that you have 6 value during either the tokenizing or casting steps.